src/MongoDB.Driver.Core/Core/Connections/TcpStreamFactory.cs

```diff
@@ -249,7 +249,16 @@ private Socket CreateSocket(EndPoint endPoint)
                 addressFamily = _settings.AddressFamily;
             }
 
-            return new Socket(addressFamily, SocketType.Stream, ProtocolType.Tcp);
+            var socket = new Socket(addressFamily, SocketType.Stream, ProtocolType.Tcp);
```

Comment:
This is the only place we create a socket so we don't need to turn on keepAlive anywhere else.